src/main/kotlin/uk/gov/justice/digital/hmpps/dpssmoketest/service/QueueService.kt

```diff
@@ -39,10 +61,42 @@ class QueueService(
   }
 
   fun checkForEvent(eventTypeRequired: String, nomsNumber: String): Boolean {
-    return true
+    var eventFound = false
+    while (hmppsEventQueueSqsClient.countMessagesOnQueue()> 0 && !eventFound) {
+      hmppsEventQueueSqsClient.receiveMessage(
+        ReceiveMessageRequest(hmppsEventQueueUrl)
+          .withMaxNumberOfMessages(1)
+      ).messages.firstOrNull()
+        ?.also { msg ->
+          val (message, messageId, messageAttributes) = objectMapper.readValue(msg.body, HMPPSMessage::class.java)
```

Comment:
Maybe extract these lines to a extension function so you can write `msg.toHMPPSDOmainEvent()`

Reply:
good idea - will look at that in next PR